File: FEBRUARY/22-02-2026/868.cpp
class Solution {
private:
    string toBinary(int n ){
        string ans = "";
        while(n > 0){
            if(n % 2 == 1){
                ans.push_back('1');
            } else {
                ans.push_back('0');
            }
            n /= 2;
        }
        return ans; 
    }
public:
    int binaryGap(int n) {
        string s = toBinary(n);
        int last = -1;  
        int maxGap = 0;

        for (int i = 0; i < (int)s.size(); i++) {
            if (s[i] == '1') {
                if (last != -1) {
                    maxGap = max(maxGap, i - last);
                }
                last = i;
            }
        }
        return maxGap;
    }
};

File: FEBRUARY/22-02-2026/868.java
class Solution {
    public int binaryGap(int n) {
        int cur = 0;
        int prev = -1;

        int res = 0;

        while (n > 0) {
            if ((n & 1) > 0) {
                res = prev != -1 ? Math.max(res, cur - prev) : res;
                prev = cur;
            }
            cur++;
            n >>= 1;
        }

        return res;
    }
}